src/Symfony/Bridge/Twig/Tests/Extension/SecurityExtensionTest.php

```diff
@@ -60,49 +70,74 @@ public function testIsGrantedThrowsWhenFieldNotNullAndFieldVoteClassDoesNotExist
         }
 
         $user = $this->createMock(UserInterface::class);
-        $userSecurityChecker = $this->createMock(UserAuthorizationCheckerInterface::class);
-        $userSecurityChecker
-            ->expects($this->once())
-            ->method('isGrantedForUser')
-            ->with($user, 'ROLE', $expectedSubject)
-            ->willReturn(true);
+        $securityChecker = $this->createMockAuthorizationChecker();
 
-        $securityExtension = new SecurityExtension(null, null, $userSecurityChecker);
+        $securityExtension = new SecurityExtension($securityChecker);
         $this->assertTrue($securityExtension->isGrantedForUser($user, 'ROLE', $object, $field));
+        $this->assertSame($user, $securityChecker->user);
+        $this->assertSame('ROLE', $securityChecker->attribute);
+
+        if (null === $field) {
+            $this->assertSame($object, $securityChecker->subject);
+        } else {
+            $this->assertEquals($expectedSubject, $securityChecker->subject);
+        }
+    }
+
+    public static function provideObjectFieldAclCases()
+    {
+        return [
+            [null, null, null],
+            ['object', null, 'object'],
+            ['object', false, new FieldVote('object', false)],
+            ['object', 0, new FieldVote('object', 0)],
+            ['object', '', new FieldVote('object', '')],
+            ['object', 'field', new FieldVote('object', 'field')],
+        ];
     }
 
     public function testIsGrantedForUserThrowsWhenFieldNotNullAndFieldVoteClassDoesNotExist()
     {
-        if (!class_exists(UserAuthorizationCheckerInterface::class)) {
+        if (!interface_exists(UserAuthorizationCheckerInterface::class)) {
             $this->markTestSkipped('This test requires symfony/security-core 7.3 or superior.');
         }
 
-        $securityChecker = $this->createMock(UserAuthorizationCheckerInterface::class);
+        $securityChecker = $this->createMockAuthorizationChecker();
 
-        ClassExistsMock::register(SecurityExtension::class);
         ClassExistsMock::withMockedClasses([FieldVote::class => false]);
 
         $this->expectException(\LogicException::class);
-        $this->expectExceptionMessageMatches('Passing a $field to the "is_granted_for_user()" function requires symfony/acl.');
+        $this->expectExceptionMessage('Passing a $field to the "is_granted_for_user()" function requires symfony/acl.');
 
-        $securityExtension = new SecurityExtension(null, null, $securityChecker);
-        $securityExtension->isGrantedForUser($this->createMock(UserInterface::class), 'object', 'bar');
+        $securityExtension = new SecurityExtension($securityChecker);
+        $securityExtension->isGrantedForUser($this->createMock(UserInterface::class), 'ROLE', 'object', 'bar');
     }
 
-    public static function provideObjectFieldAclCases()
+    private function createMockAuthorizationChecker(): AuthorizationCheckerInterface&UserAuthorizationCheckerInterface
```

Comment:
```suggestion
    private function createDummyAuthorizationChecker(): AuthorizationCheckerInterface&UserAuthorizationCheckerInterface
```

No mock is created in this method

Reply:
it's still a mock to me, conceptually